data-structures/LinkedList.py

```diff
@@ -1,121 +1,110 @@
 class Node:
-    def __init__(self, value, next_node=None):
+    def __init__(self, value, next=None):
         self.value = value
-        self.next = next_node
+        self.next = next
 
 
 class LinkedList:
     def __init__(self):
         self.head = None
+        self.tail = None
         self.total = 0
 
     def add(self, value):
-        """ Add to the end of the list """
-        new_element = Node(value)
-        if not self.head:
-            self.head = new_element
-        else:
-            current = self.head
-            while current.next:  # while my current has a next Node
-                current = current.next
-            # add the element to the last node in the list
-            current.next = new_element
-
-        self.total += 1  # update total
+        """Add an element to the end of the list."""
+        new_item = Node(value)
+        if not self.head:  # If there is no element in the list
+            self.head = new_item
+            self.tail = new_item
+        else:              # If there are elements in the list
+            self.tail = new_item
```

Comment:
what about the previous last tail? how do i get to this new item if only self.tail point to it? You're forgetting an update to the old tails.next before updating tail.